### PR TITLE
Beta Bug Fixes: Bottom Bar, Semester View

### PR DESCRIPTION
**Pull request by benjamin-shen**
### Summary 

This pull request fixes the following beta bugs.

- [x] Make bottom bar tabs not jump - [Monday](https://courseplan.monday.com/boards/433956786/pulses/551457755)
- [x] Center modal buttons - [Monday](https://courseplan.monday.com/boards/433956786/pulses/679957248)
- [x] Move confirmation message to center of semester view - [Monday](https://courseplan.monday.com/boards/433956786/pulses/550767697)
- [x] Fix new semester dropdown height for duplicate semester - 
image
- [x] Some other small html/css fixes, including confirmation message width and new semester dropdown overflow-x

### Test Plan 

Center modal buttons:
image
image
image
Duplicate semester dropdown:
image
Center confirmation message:
image
image

Most of the other changes are UX-related, please test them on the temporary dev deployment. Especially for "Make bottom bar tabs not jump" (...ea90202), let me know if I missed any edge cases.

Old:
image
New (shifted arrow):
image

### Notes
- The confirmation message is styled in a hacky way (the side panel width is manually coded in). If anyone can figure out how to set a child's y-position fixed to the window (but x-position relative to its parent), let me know!

### CODE REVIEW

**Comment by hanywang2** [comment]
Just a super nitpick, but can we have the title of the PR's be a little more descriptive next time? Bug Fixes can be anything

File: src/components/BottomBarTab.vue [hunk 1]
```diff
@@ -1,11 +1,9 @@
...
-  <div class="bottombartab" :style="{ background: `#${color}` }" :class="{ inactive: !isFirstTab}" @click="bottomBarTabToggle(courseObj)">
-    <div class="bottombartab-wrapper" @click="toggleFromTab">
-      <img v-if="isFirstTab && !isExpanded" class="bottombartab-arrow" src="@/assets/images/uparrow-white.svg"/>
-      <img v-if="isFirstTab && isExpanded" class="bottombartab-arrow" src="@/assets/images/downarrow-white.svg"/>
+  <div class="bottombartab" :style="{ background: `#${color}` }" :class="{ inactive: !isBottomCourseFocus}" @click="bottomBarTabToggle(courseObj)">
+    <div class="bottombartab-wrapper">
       <div class="bottombartab-name">{{subject}} {{number}}</div>
     </div>
-    <img class="bottombartab-delete" src="@/assets/images/x-white.svg" @click="deleteBottomTab(subject, number)"/>
+    <img class="bottombartab-delete" src="@/assets/images/x-white.svg" @click.stop="deleteBottomTab(courseObj)"/>
```
Comment (willespencer) [inline src/components/BottomBarTab.vue:6]: Haven't seen this .stop before and struggling to understand from the docs but am curious. What does this do?

File: src/components/BottomBarTabView.vue [hunk 2]
```diff
@@ -84,10 +72,14 @@ export default {
...
     },
 
-    deleteBottomTab(subject, number) {
-      for (let i = 0; i < this.bottomCourses.length; i += 1) {
-        if (this.bottomCourses[i].subject === subject && this.bottomCourses[i].number === number) {
+    deleteBottomTab(courseObj) {
+      let focusedCourse = this.bottomCourses[this.bottomCourseFocus];
+
+      let i = 0;
+      for (; i < this.bottomCourses.length; i += 1) {
```
Comment (willespencer) [inline src/components/BottomBarTabView.vue:79]: Any reason the let is not in the first part of the for loop here? Didn't know this was possible haha.

File: src/components/Confirmation.vue [hunk 3]
```diff
@@ -8,9 +8,9 @@ method - unsure what the right move is right now
       <img class="confirmation-icon" src="../assets/images/checkmark.svg" />
     </div>
     <div class="confirmation-text">{{ text }}</div>
-    <div class="confirmation-right">
-      <!-- <span class="undo-buttonText">{{ undo }}</span> -->
-    </div>
+    <!-- <div class="confirmation-right">
```
Comment (willespencer) [inline src/components/Confirmation.vue:11]: You can honestly go ahead and delete this instead of just commenting out a better part of it. These undo buttons are in the designs but I don't think we're planning on developing them anytime soon.

File: src/components/Confirmation.vue [hunk 4]
```diff
@@ -73,10 +77,7 @@ export default {
 
     color: #7B7D7E;
 
-    margin-left: 8px;
-    margin-right: 16px;
-    margin-top: 13px;
-    margin-bottom: 12px;
+    margin: 14px 16px 12px 16px;
```
Comment (willespencer) [inline src/components/Confirmation.vue:80]: Nice 😄

**Review by willespencer — CHANGES_REQUESTED** [review]
Nice job Ben! The code looks good to me, just left a few comments!

I was also testing all of your stuff out on the deployment and noticed a few small things.
- The new tab functionality seems almost perfect! The only edge case I've found that acts a bit strange is changing the color of a course seems to auto-select the first tab. I think it would be better if changing the color of a course did not change which tab was selected at all?
- The bottom border of the two input boxes on the new semester modal seem to be cut off (when FA20 is not already in your plan). Could you make sure these borders show up?
- Bit nitpick, but I think the "Add" text on the new course modal is still a bit off center, could an `display: flex; align-items: center;` fix that?

If you just resolve these small things I think this will be good to go!

File: src/components/BottomBarTab.vue (hunk 1 above)
Comment (benjamin-shen) [reply src/components/BottomBarTab.vue:6]: Haha I looked it up too. It's a shorthand for calling `e.stopPropagation()`, which stops the click event from propagating up to any parents. https://flaviocopes.com/vue-events/#event-modifiers

So in this case, if I want to delete a bottom bar tab (by clicking the x-icon), it won't call the function normally called from clicking the tab itself (`bottomBarTabToggle(courseObj)`).
You can actually observe this bug in production right now; if you try to delete a tab besides the first tab, it switches the focus to another course.

File: src/components/BottomBarTabView.vue (hunk 2 above)
Comment (benjamin-shen) [reply src/components/BottomBarTabView.vue:79]: It's so I can use `i` in line 101 further down in the function:
```
this.bottomBarTabToggle(this.bottomCourses[i]);
```
Although if it makes more sense or is more readable, I can put `i` back in the for loop and use another variable.

Comment (willespencer) [reply src/components/BottomBarTabView.vue]: Nope this is fine! Just took me a second to figure it out 😄

**Review by willespencer — APPROVED** [review]
Looks great! You can merge it in whenever 😄
